### PR TITLE
Alpha filter

### PR DESCRIPTION
**Pull request by hunse**
Add Alpha filter implementation

### CODE REVIEW

**Review by drasmuss — APPROVED** [review]
Made various fixups, more details in the commit messages. Let me know if those changes look good to you, then will merge!

**Comment by hunse** [comment]
The problem with just using one step in `test_lowpass_trainable` is that all the system really needs to learn is the initial filter state, it doesn't need to learn `tau`. I set `trainable=False` for `tau` and found that this test still passes.

I would recommend using multiple steps and checking explicitly that `tau` has changed just to make sure it's training properly, as I did in the modified test. Unfortunately, that probably means using looser tolerances, because once it's optimizing over multiple timesteps it will choose a larger `initial_value` so that after the initial value decays for a few timesteps, it's still near the target value of 1.

It would be nice if there was a way we could have really tight tolerances and be sure that `tau` is training properly. The only way I can think to do that would be to have the target be a lowpass-filtered version of the input, with e.g. a `tau` that's twice as long as the initial `tau`. That way, the target should be perfectly learnable. (In the alpha filter test, I just use all ones, which it can never learn perfectly since it would need an infinite tau; hence the looser tolerances.)

**Comment by drasmuss** [comment]
Updated the test in the other PR (as it worked better after those changes), see https://github.com/nengo/keras-spiking/pull/5/commits/9decbd0f5c0bdef39475d61a372fc9e43f4334b3

File: keras_spiking/layers.py [hunk 1]
```diff
@@ -7,7 +7,203 @@
...
+
+    def reset_states(self, states=None):
+        """
+        Reset the internal state of the layer (only necessary if ``stateful=True``).
+
+        Parameters
+        ----------
+        states : `~numpy.ndarray`
+            Optional state array that can be used to override the values returned by
+            `.SpikingActivationCell.get_initial_state`.
```
Comment (hunse) [inline keras_spiking/layers.py:184]: This should be ` ``cell.get_initial_state`` ` or something like that.

**Comment by hunse** [comment]
The new test looks good. So I'm fine to merge this as-is (with the minor fix to the docstring I mentioned above), and then the test will get fixed when we merge #5.

Also, it might be a good idea to double-check the docstrings for the base classes, since I obviously didn't do a great job updating them when I reduced the code redundancy.

**Pull request merged**